haystack/document_stores/deepsetcloud.py

```diff
@@ -122,6 +143,9 @@ def get_all_documents(
             "`get_all_documents()` can get very slow and resource-heavy since all documents must be loaded from Deepset Cloud. "
             "Consider using `get_all_documents_generator()` instead."
         )
+        if not self.index_exists:
+            return []
+
```

Comment:
We don't need that I guess as get_all_documents_generator already handles the non existing index case.